app/services/membership/withdraw_service.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+class Membership::WithdrawService
+  attr_reader :group, :user
+
+  class MembershipMustHaveDefaultPost < StandardError; end
+
+  def initialize(group, user)
+    @group = group
+    @user = user
+  end
+
+  def self.call(group, user)
+    new(group, user).perform
+  end
+
+  def perform
+    membership = user.membership_for(group)
+    raise MembershipMustHaveDefaultPost unless membership.has_post?(PostType::DEFAULT_POST_ID)
+
+    notifications_from_user_to_group = group.leader.user
+                                                   .notifications.where(notifier_id: user.id,
+                                                                        key: 'membership.create')
+                                                   .select{ |n| n.notifiable.group == group }
+    membership.destroy!
+
+    invalid_notification = notifications_from_user_to_group.find{ |n| !n.reload.notifiable.present? }
```

Comment:
I think this can be removed with the suggested changes.

side-note: `!anything.present?` can be written `anything.blank?`